controllers/argocd/keycloak.go

```diff
@@ -1130,9 +1226,13 @@ func deleteKeycloakConfigForOpenShift(cr *argoprojv1a1.ArgoCD) error {
 		cr.Name, cr.Namespace))
 
 	oa := getOAuthClient(cr.Namespace)
-	err = oauth.OAuthClients().Delete(context.TODO(), oa, deleteOptions)
-	if err != nil {
-		return err
+
+	_, err = oauth.OAuthClients().Get(context.TODO(), oa, metav1.GetOptions{})
```

Comment:
What is the reason to do a "get" before calling delete?

Reply:
OAuthClient configuration does not get deleted from previous instances occasionally. I want to re-use this function to handle such cases.

Reply:
I noticed this issue once a very long time back but I thought it is issue with my configuration, I noticed this again recently. I did not find any issue on OpenShift API.

So to be 100% sure, I added this double check.